sktime/performance_metrics/forecasting/tests/test_all_metrics_forecasting.py

```diff
@@ -186,3 +186,25 @@ def test_metric_weights(self, estimator_instance):
         # wt_metr = metric(sample_weight=wts)
         # res_wt = wt_metr(y_true, y_pred)
         # assert np.allclose(res_wt, metric_obj(y_true, y_pred, sample_weight=wts))
+
+    def test_evaluate_by_index(self, estimator_instance):
```

Comment:
hm, looking at the tests, I would have added it at the end of `test_metric_output_by_instance`, as follows:

* clone the metric used above and `set_params` the `by_index`, or equivalent copy/set logic
* execute the `__call__`
* check that it is identical with the return (using `pandas` equals)

This works because we checked all conventions on `evaluate_by_index` above already, and we now only need to additionally check that we get the same from the call.